tempodb/blocklist/poller.go

```diff
@@ -85,13 +121,37 @@ func (p *Poller) Do() (PerTenant, PerTenantCompacted, error) {
 }
 
 func (p *Poller) pollTenant(ctx context.Context, tenantID string) ([]*backend.BlockMeta, []*backend.CompactedBlockMeta, error) {
+	// if we're not building the tenant index then attempt to pull it
+	if !p.buildTenantIndex() {
+		metricTenantIndexBuilder.Set(0)
+
+		i, err := p.reader.TenantIndex(ctx, tenantID)
+		if err == nil {
+			metricTenantIndexAgeSeconds.WithLabelValues(tenantID).Set(float64(time.Since(i.CreatedAt) / time.Second))
+			level.Info(p.logger).Log("msg", "successfully pulled tenant index", "tenant", tenantID, "createdAt", i.CreatedAt, "metas", len(i.Meta), "compactedMetas", len(i.CompactedMeta))
+			return i.Meta, i.CompactedMeta, nil
+		}
+
+		metricTenantIndexErrors.WithLabelValues(tenantID).Inc()
+
+		// we had an error, do we bail?
+		if !p.cfg.PollFallback {
+			return nil, nil, err
+		}
+
+		// log error and keep on keeping on
+		level.Error(p.logger).Log("msg", "failed to pull bucket index for tenant. falling back to polling", "tenant", tenantID, "err", err)
```

Comment:
I think the very first time the tenant index is being created, this log line will be hit. The tenant index will not be generated, and all components will be required to poll like they always do.. Should this be an "error" level log?

Reply:
Agree there will be a rush of this error the first time that you upgrade to a version that supports the tenant index, but during normal operation I would consider this an error and think it should remain at this level.